R/escape.R

```diff
@@ -229,13 +228,37 @@ sql_escape_logical <- function(con, x) {
   UseMethod("sql_escape_logical")
 }
 
+#' @keywords internal
+#' @export
+sql_escape_Date <- function(con, x) {
+  UseMethod("sql_escape_Date")
+}
+
+
+#' @keywords internal
+#' @export
+sql_escape_POSIXt <- function(con, x) {
+  UseMethod("sql_escape_POSIXt")
+}
+
 # DBIConnection methods --------------------------------------------------------
 
 #' @export
 sql_escape_string.DBIConnection <- function(con, x) {
   dbQuoteString(con, x)
 }
 
+#' @export
+sql_escape_Date.DBIConnection <- function(con, x) {
+  dbQuoteString(con, x)
```

Comment:
DBI has this, DBI backends can override.

```suggestion
  dbQuoteLiteral(con, x)
```

Reply:
I'm sorry, could you explain a bit what would be the benefit of this change? dbQuoteLiteral seems to format a date as something like '2020-04-15 UTC' which does not work at least in Athena (don't have other databases up right now to test against). So if I would use dbQuoteLiteral, I would still need to first cast the date as character.

Surely I'm just missing the point here, so please help me out :)